### PR TITLE
Remove cozy client dependency

### PR DESCRIPTION
**Pull request by kosssi**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -43,8 +43,17 @@ function subscribeWhenReady(doctype, socket) {
...
+function getWebsocketProtocol(url) {
+  const httpsRegexp = new RegExp(`^(https:\/{2})`)
+  return url.match(httpsRegexp) ? 'wss' : 'ws'
+}
+
+function getDomainFromUrl(url) {
+  const domainRegexp = new RegExp(`^https?:\/{2}(.*)$`)
+  const m = url.match(domainRegexp)
+  if (m) {
+    m[1]
```
Comment (gregorylegarec) [inline src/index.js]: missing `return`

**Comment by sebn** [comment]
> I feel that it's a little bit constraining to pass url and token at every subscribe call. May be we could initialize the realtime once as we do for the other cozy clients.

:+1: `url` and `token` always *travel* together, so they should probably be part of the same *config* object, so we may reuse it, don't need to remember params order, and don't need to change function calls in case *config* gets a new field.

Although I would prefer one of the following solutions:

1. Just pass the *config* objet to function calls, e.g. `realtime.subscribe(config, 'io.cozy.files')` (may be a first, easier to implement step)
2. Instanciate the realtime, e.g. `const realtime = Realtime.init()` (so consumer code doesn't need to know whether the result is built using a class or a closure. Module vs variable names always bother me in js, hence the capital `R`, but any other name would be fine by me)

In both case, we don't expose functions until they can actually be used so we don't get unexpected errors when calling them (a designer once gave me this example: "Tea or coffee? Oh sorry we don't have any of them...").

**Comment by ptbrowne** [comment]
I think `init`ing is the right solution as it is cumbersome to remember to pass the config object.

**Comment by gregorylegarec** [comment]
So I suggest to:
* First use a config object to get rid of cozy-client-js quickly and keep almost the same method signature
* Second aim to use an `init()` method in a near future, let's create an issue.

What do you think guys ?

**Comment by CPatchane** [comment]
Completely agree about the config object usage and the `init()` method 👍 🌮

**Comment by CPatchane** [comment]
Some news about this PR @kosssi? I'll wait for this change before moving it into `cozy-libs`.

**Comment by CPatchane** [comment]
Thanks @kosssi 🍶

**Pull request merged**